diff --git a/viewflow/workflow/activation.py b/viewflow/workflow/activation.py
index 7f493e7..d36e9c5 100644
--- a/viewflow/workflow/activation.py
+++ b/viewflow/workflow/activation.py
@@ -195,12 +195,12 @@ class Activation:
         Returns:
             Activation: The newly created activation instance.
         """
-        flow_class = flow_task.flow_class
         task = flow_class.task_class(
             process=prev_activation.process,
             flow_task=flow_task,
             token=token,
         )
+        print(f"[create_task] token={token}, owner_process_id={task.process_id}")
         task.data = data if data is not None else {}
         task.seed = seed
         task.save()
diff --git a/viewflow/workflow/nodes/split.py b/viewflow/workflow/nodes/split.py
index a325676..74b6a46 100644
--- a/viewflow/workflow/nodes/split.py
+++ b/viewflow/workflow/nodes/split.py
@@ -21,12 +21,10 @@ class SplitActivation(Activation):
 
     @Activation.status.super()
     def activate(self):
-        """Calculate nodes list to activate."""
         for node, cond, data_source in self.flow_task._branches:
             if cond and not cond(self):
                 continue
-            else:
-                self.next_tasks.append((node, None))
+            self.next_tasks.append((node, data_source))
 
         if not self.next_tasks:
             raise FlowRuntimeError(
@@ -126,16 +124,11 @@ class Split(
         return self._activate_next
 
     def Next(self, node, case=None):
-        """Node to activate if condition is true.
-
-        :param cond: Callable[activation] -> bool
-
-        """
         self._activate_next.append((node, case, None))
         return self
 
     def Always(self, node):
-        return self.Next(node)
+        self._activate_next.append((node, None, None))
 
 
 class SplitFirstActivation(SplitActivation):
